tools/bowtie2/bowtie2_wrapper.xml

```diff
@@ -145,18 +145,19 @@
         
         #elif str( $analysis_type.analysis_type_selector ) == "cline":
             ${analysis_type.cline}
-        #end if    
+        #end if
         
+        -S "galaxy_bowtie_2_output.sam" &amp;&amp;
+
         ## view/sort and output BAM file
-        | samtools view -Su - | samtools sort -o - - > $output
+        ( samtools view -Su "galaxy_bowtie_2_output.sam" | samtools sort -o - - > $output )
 
         ## rename unaligned sequence files
         #if $library.type == "paired" and $output_unaligned_reads_l and $output_unaligned_reads_r:
-            #set left  = str($output_unaligned_reads_l).replace( '.dat', '.1.dat' )
-            #set right = str($output_unaligned_reads_l).replace( '.dat', '.2.dat' )
-        
-            ; mv $left $output_unaligned_reads_l;
-            mv $right $output_unaligned_reads_r
+            #set left  = str($output_unaligned_reads_l)[:-3] + '1.dat'
+            #set right = str($output_unaligned_reads_l)[:-3] + '2.dat'
```

Comment:
This looks ok. Bowtie must be spliting on the ext and then adding the 1/2:
-rw-r--r--    1 dan  staff   178B Mar 20 08:46 dataset_186.1.dat
-rw-r--r--    1 dan  staff   178B Mar 20 08:46 dataset_186.2.dat
-rw-r--r--    1 dan  staff     0B Mar 20 08:46 dataset_186.dat

It likely looks less odd when the passed in output filename is e.g., unaligned.fa.

I don't like that it works on the assumption that the output files all end in .dat (we should probably do a splitext to figure it out), but that isn't a change in behavior.